app/src/main/java/com/realityexpander/tasky/agenda_feature/domain/AgendaItem.kt

```diff
@@ -1,32 +1,151 @@
 package com.realityexpander.tasky.agenda_feature.domain
 
+import android.os.Parcelable
+import com.realityexpander.tasky.core.util.UuidStr
+import kotlinx.parcelize.Parcelize
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
 import java.time.LocalDateTime
+import java.util.*
 
 // NOTE: Skeleton data structures for the Agenda feature
 // **temporary** for UI development
 
+@OptIn(ExperimentalSerializationApi::class)
+@Serializer(forClass = LocalDateTime::class)
+object LocalDateTimeSerializer : KSerializer<LocalDateTime> {
+    override val descriptor = PrimitiveSerialDescriptor("LocalDateTime", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): LocalDateTime {
+        return LocalDateTime.parse((decoder.decodeString()))
+    }
+
+    override fun serialize(encoder: Encoder, value: LocalDateTime) {
+        encoder.encodeString(value.toString())
+    }
+}
```

Comment:
This is against the purpose of domain, since it reveals implementation details (what kind of JSON/serialization lib you use). Why not use Parcelable + `@Parcelize`?

Reply:
I have to use both `@Parcelize` and `@Serialize`. The `SavedStateHandle` uses `@Parcelize` under the hood, and the navigation library im using uses `@Serialize`. 

Where should I move the LocalDateTimeSerializer to? Which package?